### PR TITLE
Feature/pt 14/api documentation

### LINKED ISSUE #14: Document API with swagger
### Setup Guide
- https://blog.logrocket.com/documenting-express-js-api-swagger/

### Packages
- https://www.npmjs.com/package/swagger-jsdoc 
- https://www.npmjs.com/package/swagger-ui-express

### PR DESCRIPTION
**Pull request by samresh9**
Fixes #14 
Setup Swagger
[Swagger Documentation](https://node-protectedtext-pr-21.onrender.com/)

### CODE REVIEW

File: src/models/notesModel.js [hunk 1]
```diff
@@ -2,18 +2,18 @@ const mongoose = require("mongoose");
...
+    encryptedContent: {
       type: String,
       required: [true, "A note content is required"],
     },
-    hashContent: {
+    hash: {
       type: String,
-      required: [true, "A hash content is required"],
+      required: [true, "A hash of content is required"],
     },
```
Comment (therj) [inline src/models/notesModel.js:17]: This is an interesting problem to have!

Change in schema makes existing data useless, or worse conflicting.
We already have notes in db - we do not want to lose that. 
Create a migration script that moves existing data into new format, put migration scripts in a separate folder - {projectRoot}/migrations/YYYYMMDD_script_name.js.

**Review by therj — CHANGES_REQUESTED** [review]

File: src/models/notesModel.js (hunk 1 above)
Comment (samresh9) [reply src/models/notesModel.js:17]: I faced the problem but had no I idead about creating a separate file for migration I ran command on  mongoshell for $rename fieldName.
db.notes.updateMany({},{$rename:{'noteId':'id', 'content':'encryptedContent', 'hashContent':'hash'}})
Is this a wrong approach?

Comment (therj) [reply src/models/notesModel.js:17]: The script should be tested in dev environments, and everyone should be able to replicate the data migration anywhere and everywhere.

Create the script.. Deploy the old commit, create notes, try migrating again. Now, you've even more interesting problem as you will have a mix of old and new structure of data. 🧐

File: src/routes/notesRoutes.js [hunk 2]
```diff
@@ -1,3 +1,229 @@
...
+
+/**
+ * @swagger
+ * tags:
+ *   name: Notes
+ *   description: API for creating, updating, and getting notes
+ */
+/**
+ * @swagger
+ * /api/notes:
```
Comment (therj) [inline src/routes/notesRoutes.js:118]: Let's keep /api/ out of route name on swagger.

File: src/routes/notesRoutes.js [hunk 3]
```diff
@@ -1,3 +1,229 @@
...
+ *                 - type: "field"
+ *                   msg: "Note unique id is required"
+ *                   path: "id"
+ *                   location: "body"
+ *       500:
+ *         $ref : '#/components/responses/InternalServerError'
+ */
+/**
+ * @swagger
+ * /api/notes/{id}:
```
Comment (therj) [inline src/routes/notesRoutes.js:189]: +1
Let's keep /api/ out of route name on swagger.

File: src/models/notesModel.js (hunk 1 above)
Comment (samresh9) [reply src/models/notesModel.js:17]: Do migrating means transforming old format to new and create new data or modifying the old data?
and by default  _v: 0 is present for both new and old format. Should I manually change the _v vlaue when creating new schema data?

Comment (therj) [reply src/models/notesModel.js:17]: New data can have _v: 1,  old can have _v: 0. Both can co-exist. This is just one way to do things, but you can just write script to migrate based on field name, if doc.noteId, replace the document - else (node.id) let it be.

**Review by therj — APPROVED** [review]

**Pull request merged**